File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] day = new int[temperatures.length];
        Stack<Integer> st = new Stack<>();
        for(int i=0; i<temperatures.length; i++)
        {
            while(!st.isEmpty() && temperatures[i] > temperatures[st.peek()])
            {
                int j = st.peek();
                st.pop();
                day[j] = i - j;
            }
            st.push(i);
        }
        return day;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> max(temperatures.size());
        stack<int> st;
        for(int i=0; i<temperatures.size(); i++)
        {
            while(!st.empty() && temperatures[i] > temperatures[st.top()])
            {
                int prev = st.top();
                st.pop();
                max[prev] = i - prev;
            }
            st.push(i);
        }
        return max;
    }
};